### PR TITLE
[#IR-125] Testing Fixes

### PR DESCRIPTION
**Pull request by chetanyakan**

### CODE REVIEW

File: src/app/retrospective/sprint-detail/sprint-detail.component.html [hunk 1]
```diff
@@ -157,7 +157,7 @@ <h4 class="title">
                             </div>
                         </mat-tab>
                         <mat-tab label="Task Summary">
-                            <div *ngIf="sprintDetails.LastSyncedAt && sprintDetails.SyncStatus !== syncStates.SYNCING; else not_synced">
+                            <div *ngIf="sprintDetails.LastSyncedAt && sprintDetails.SyncStatus !== syncStates.SYNCING && sprintDetails.SyncStatus !== syncStates.QUEUED; else not_synced">
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-detail/sprint-detail.component.html]: use `indexOf` to check for sprint sync status

File: src/app/retrospective/sprint-detail/sprint-detail.component.ts [hunk 2]
```diff
@@ -65,10 +67,19 @@ export class SprintDetailComponent implements OnInit, OnDestroy  {
                 }, delay);
             });
         this.refresh$.next();
+        Observable.interval(5000)
```
Comment (anmolghosh) [inline src/app/retrospective/sprint-detail/sprint-detail.component.ts]: move this to a local or global constant

File: src/app/retrospective/sprint-detail/sprint-detail.component.ts [hunk 3]
```diff
@@ -89,7 +100,7 @@ export class SprintDetailComponent implements OnInit, OnDestroy  {
                 this.selectedValue = ACTIONABLE_SPRINT_STATES[this.sprintStatus];
                 this.sprintDays = this.utils.workdayCount(response.data.StartDate, response.data.EndDate);
                 if (this.sprintDetails.SyncStatus === SPRINT_SYNC_STATES.SYNCING ||
```
Comment (anmolghosh) [inline src/app/retrospective/sprint-detail/sprint-detail.component.ts]: this seems buggy. Please revisit

File: src/app/shared/utils/utils.service.ts [hunk 4]
```diff
@@ -8,11 +8,14 @@ export class UtilsService {
     constructor(private datePipe: DatePipe) { }
 
     getApiErrorMessage(response): string {
-        if (!response.data.error) {
+        if (response.data.error) {
+            const message = response.data.error.trim();
```
Comment (anmolghosh) [inline src/app/shared/utils/utils.service.ts]: can't we use `_.upperFirst`?

File: src/constants/app-constants.ts [hunk 5]
```diff
@@ -47,6 +47,7 @@ export const APP_ROUTE_URLS = {
     teamFeedbackList: 'team/feedbacks',
     retrospectiveList: 'retrospectives',
     sprintDetails: 'retrospectives/:retrospectiveID/sprint/:sprintID',
+    retrospectiveDetail: 'retrospectives/:retrospectiveID',
```
Comment (anmolghosh) [inline src/constants/app-constants.ts]: club the urls

File: src/constants/app-constants.ts [hunk 6]
```diff
@@ -103,28 +104,29 @@ export const API_RESPONSE_MESSAGES = {
...
+    goalUnResolveFailed: `Failed to un-resolve the goal!`,
+    sprintHighlightsGetError: 'Couldn\'t get the sprint highlights!',
+    sprintHighlightsUpdateError: `Couldn't update sprint highlight!`,
+    sprintHighlightsUpdateSuccess: 'Successfully updated sprint highlight!',
+    sprintHighlightsAddError: `Couldn't add a sprint highlight!`,
+    sprintHighlightsAddSuccess: 'Successfully added a sprint highlight!',
+    sprintNotesGetError: 'Couldn\'t get the sprint notes!',
+    sprintNotesUpdateError: `Couldn't update sprint note!`,
+    sprintNotesUpdateSuccess: 'Successfully updated sprint note!',
+    sprintNotesAddError: `Couldn't add a sprint note!`,
```
Comment (anmolghosh) [inline src/constants/app-constants.ts]: keep things consistent, don't use both \`\` and ''

**Pull request merged**